### PR TITLE
Add last dispensed medication to vih dashboard

### PR DESCRIPTION
**Pull request by ljoedavessenn**
@mogoodrich suggested that I added two set of information for the regimen :

- The last dispensed medication set (ARV1 -ARV2- ARV3)
- The last prescribed medication set from the orders.

Here I added the last dispensed set.

### CODE REVIEW

File: configuration/pih/statusData/hiv/hivStatuses.yml [hunk 1]
```diff
@@ -43,4 +43,10 @@
...
   statusDataQuery: "lastCd4.sql"
   valueExpression: "#if($cd4)$cd4 $fn.translate('general.onDate') $fn.formatDate($cd4Date)#{else}$fn.translate('zl.status.lastCd4.none')#end"
-  formatExpression: ""
\ No newline at end of file
+  formatExpression: ""
+
+- id: "lastDispensedMedication"
+  labelCode: "zl.status.regimen.lastDispensedMedication"
+  statusDataQuery: "lastDispensedMedication.sql"
+  valueExpression: "#if(!$arv1DrugName && $arv2DrugName && $arv3DrugName) $fn.translate('zl.status.regimen.lastDispensedMedication.none') #end  #if($arv1DrugName) $arv1Name : $arv1DrugName ($arv1DrugQuantity) <br/> #end #if($arv2DrugName) $arv2Name : $arv2DrugName ($arv2DrugQuantity) <br/> #end #if($arv3DrugName) $arv3Name : $arv3DrugName ($arv3DrugQuantity) #end"
```
Comment (mogoodrich) [inline configuration/pih/statusData/hiv/hivStatuses.yml:51]: Shouldn't there be a  "!" in front of all three drugs here? it looks from the display text that you are trying to test if all of them are null/empty?

**Review by mogoodrich — COMMENTED** [review]
Thanks @ljoedavessenn !  The general format looks good, though I had one comment on the "valueExpression".

I will defer to @dmdesimone for SQL logic to retrieve the three drugs.  Thanks!

File: configuration/messageproperties/messages_zl_fr.properties [hunk 2]
```diff
@@ -29,6 +29,9 @@ zl.status.hivTb.inhCompletionDate=Date d'achèvement de l'INH
 zl.status.hivTb.inhComplete=Complété
 zl.status.hivTb.inhNotComplete=Non complété
 zl.status.hivTb.inhNotStarted=Non initié
+zl.status.regimen.lastPrescribedMedication=Dernier médicament prescrit
+zl.status.regimen.lastDispensedMedication=Dernier médicament distribué
+zl.status.regimen.lastDispensedMedication.none=Aucun
```
Comment (lnball) [inline configuration/messageproperties/messages_zl_fr.properties:34]: Use an existing message property -- pihcore.none.label .  Not necessary to create anything new for "none".

File: configuration/pih/statusData/hiv/lastDispensedMedication.sql [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+into @dispensingConstruct, @medicationCategory, @arv1, @arv2, @arv3, @medicationName, @medicationQuantity
+;
+
+
+select concept_name(@arv1, 'en'), concept_name(@arv2, 'en'), concept_name(@arv3, 'en')
+into  @arv1Name, @arv2Name, @arv3Name;
+
+
+# Get the latest encounter with ARV1 or ARV2 or ARV3
+SELECT o.encounter_id  into @encounterId
```
Comment (lnball) [inline configuration/pih/statusData/hiv/lastDispensedMedication.sql:20]: I think you want to limit the query to encounter_type of "HIV drug dispensing".

**Review by lnball — COMMENTED** [review]
Get in touch with any questions.  Sorry that I didn't respond quicker to Teams.

**Review by lnball — APPROVED** [review]
Looks good, but I didn't test the code.  Push your changes and feel free to assign testing to me.

**Pull request merged**